jina/peapods/gateway.py

```diff
@@ -114,6 +114,14 @@ def handle(self, msg: 'jina_pb2.Message'):
             try:
                 self._request = getattr(msg.request, msg.request.WhichOneof('body'))
                 self._message = msg
+                # Check if pod had thrown a run-time exception by checking envelope's status and error message
+                if msg.envelope.status == jina_pb2.Envelope.Status.ERROR:
+                    error_message = f'pod threw run time exception {msg.envelope.error_message}'
+                    self.logger.info(error_message)
+                    # extract error message from envelope, remove envelope, send back request and error message
+                    return msg.request, msg.envelope.error_message
```

Comment:
Note that the gateway always removes the `Envelope` as those information is often not interested to the client. Except this `error_message`, the client has to know it. So maybe `error_message` should be carried as a part of `Request`.

Reply:
changed the Request proto to include the error message